### PR TITLE
Allow client to suppress automatic logging by flag

### PR DESCRIPTION
**Pull request by maxponte**

### CODE REVIEW

File: flag_set.go [hunk 1]
```diff
@@ -95,6 +102,16 @@ func (efs *FlagSet) Disable(flag Flag) {
 	efs.flags[flag] = false
 }
 
+// Hide disallows automatic logging for all events under a flag
+func (efs *FlagSet) Hide(flag Flag) {
```
Comment (wcharczuk) [inline flag_set.go]: omit in favor of the `Hide(...)` variadic

File: flag_set.go [hunk 2]
```diff
@@ -95,6 +102,16 @@ func (efs *FlagSet) Disable(flag Flag) {
...
+	efs.shouldNotWrite[flag] = true
+}
+
+// Show allows automatic logging for all events under a flag
+func (efs *FlagSet) Show(flag Flag) {
```
Comment (wcharczuk) [inline flag_set.go]: ditto as `Hide`, just have a `Show(...)` variadic

File: flag_set.go [hunk 3]
```diff
@@ -79,9 +85,10 @@ func NewFlagSetFromCSV(flagCSV string) *FlagSet {
 
 // FlagSet is a set of event flags.
 type FlagSet struct {
-	flags map[Flag]bool
-	all   bool
-	none  bool
+	shouldNotWrite map[Flag]bool
```
Comment (wcharczuk) [inline flag_set.go]: can we make this `hidden`

File: environment_variables.go [hunk 4]
```diff
@@ -2,9 +2,12 @@ package logger
 
 //env var names
 const (
-	// EnvVarLogEvents is the log verbosity environment variable.
+	// EnvVarLogEvents is the event flag set environment variable.
 	EnvVarLogEvents = "LOG_EVENTS"
 
+	// EnvVarLogHiddenEvents is the set of flags that should never produce automatic output.
+	EnvVarLogHiddenEvents = "LOG_HIDDEN_EVENTS"
```
Comment (wcharczuk) [inline environment_variables.go]: I would just make this `LOG_HIDDEN` to make it shorter

File: logger.go [hunk 5]
```diff
@@ -269,7 +328,7 @@ func (l *Logger) Trigger(e Event) {
 	l.ensureInitialized()
 
 	flag := e.Flag()
-	if l.flags.IsEnabled(flag) {
+	if l.IsEnabled(flag) {
```
Comment (maxponte) [inline logger.go]: also checks nil

**Comment by wcharczuk** [comment]
lgtm

**Pull request merged**